File: packages/fluxguard-node/src/store/Store.ts
/**
 * Storage abstraction: local Map or Redis (Lua via evalScript).
 */
export interface Store {
  get(key: string): Promise<string | null>;
  set(key: string, value: string, ttlMs: number): Promise<void>;
  del(key: string): Promise<void>;
  /**
   * Execute Lua atomically (Redis). Local store may throw if used.
   */
  evalScript(script: string, keys: string[], args: (string | number)[]): Promise<unknown[]>;
}

File: packages/fluxguard-java/src/main/java/com/fluxguard/store/Store.java
package com.fluxguard.store;

import java.util.List;

public interface Store {
  String get(String key);

  void set(String key, String value, long ttlMs);

  void del(String key);

  List<Object> evalScript(String script, List<String> keys, List<String> args);
}
